### PR TITLE
Improve output when connection is not made

### PR DESCRIPTION
**Pull request by ashishpriyadarshiCIC**
Solves issue #1247

### CODE REVIEW

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde Can you please review it

File: retriever/engines/postgres.py [hunk 1]
```diff
@@ -249,11 +249,16 @@ def get_connection(self):
...
-                             user=self.opts["user"],
-                             password=self.opts["password"],
-                             database=self.opts["database"])
+        try:
+            conn = dbapi.connect(host=self.opts["host"],
+                                 port=int(self.opts["port"]),
+                                 user=self.opts["user"],
+                                 password=self.opts["password"],
+                                 database=self.opts["database"])
+        except dbapi.OperationalError as _:
```
Comment (henrykironde) [inline retriever/engines/postgres.py]: Why do we two exception handlers ?

**Comment by henrykironde** [comment]
Did you test this. And could you paste the test results

File: retriever/engines/postgres.py (hunk 1 above)
Comment (ashishpriyadarshiCIC) [reply retriever/engines/postgres.py]: I thought if there was another error it would not print the same message

Comment (henrykironde) [reply retriever/engines/postgres.py]: I think we can put it on the same line except (dbapi.OperationalError, Exception) as err

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde Here is the output :- 
Screenshot from 2019-03-12 00-18-23

**Comment by ethanwhite** [comment]
It seems like ideally we should stop as soon as the connection fails rather than continuing with the following installation steps.

**Comment by henrykironde** [comment]
Ethan is correct. If you trace the code from the get connection, you should guide the execution to return to main. So search for the functions that call get connection. and return if the connection is None. 
Try running `retriever install --debug  postgres  iris` to see the trace

**Comment by ashishpriyadarshiCIC** [comment]
I'll start working on it as I was using `raise` initially to stop.

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde I left this PR for too long, I will complete it now

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde  Should I create a new PR as this branch is too outdated ?

**Comment by henrykironde** [comment]
Sure go for it

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde Closed this PR as another PR #1432  regarding the issue had been merged.

**Pull request closed without merging**